hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSetProcessor.java

```diff
@@ -672,6 +671,37 @@ public Boolean isCached(long conglomerateId) throws StandardException {
 
     }
 
+    /**
+     * since spark has only strings, not CHAR/VARCHAR,
+     * we need to "create" CHAR/VARCHAR column out of string columns
+     * - for CHAR, we need to right-pad strings
+     * - for CHAR/VARCHAR, we need to make sure we're considering the maximum string length
+     *   note that we will use Java/Scala String length, which is measured in
+     *   UTF-16 characters, which is NOT the byte length and also NOT necessarily the character length.
+     *   e.g. the single character U+1F602 (https://www.fileformat.info/info/unicode/char/1f602/index.htm)
+     *   is encoded as 0xD83D 0xDE02 and therefore has length 2.
+     */
+    private Column convertSparkStringColToCharVarchar(Column col, DataValueDescriptor dvd, String name) {
```

Comment:
I am almost sure you don't need `col` parameter here. If you think you still need it, maybe you could consider splitting this function again such that you don't have as in/out parameter.

Reply:
the column is needed as you create the new column from the old column by transforming it with a function